xmake/core/platform/platform.lua

```diff
@@ -151,7 +151,7 @@ end
 -- get the toolchains
 function _instance:toolchains(opt)
     local toolchains = self:_memcache():get("toolchains")
-    if not toolchains then
+    if not toolchains or #toolchains == 0 then
```

Comment:
why?

Reply:
When I selected wasm as the platform, the toolchains list was empty but not nil, which did not allow me to install headeronly packages

Reply:
it should not be empty.

Reply:
I can't reproduce this. You can print debug.traceback() when itis empty. 

or revert it and open an issue first. I will merge other patches.

Reply:
I got this error before adding the line:
```
error: ...amdir\core\sandbox\modules\import\core\tool\compiler.lua:37: cannot get program for cc
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:973]:
    [...amdir\core\sandbox\modules\import\core\tool\compiler.lua:37]: in function 'load'
    [...amdir\core\sandbox\modules\import\core\tool\compiler.lua:316]:
    [@programdir\modules\package\tools\cmake.lua:886]: in function '_get_envs_for_runtime_flags'
    [@programdir\modules\package\tools\cmake.lua:928]: in function '_get_configs'
    [@programdir\modules\package\tools\cmake.lua:1234]: in function 'install'
    [...\.xmake\repositories\xmake-repo\packages\f\fmt\xmake.lua:70]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...\modules\private\action\require\impl\actions\install.lua:392]:

  => install fmt 11.0.2 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:494: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:973]:
    [...\modules\private\action\require\impl\actions\install.lua:494]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:361]:
    [...modules\private\action\require\impl\install_packages.lua:496]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:973: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
```

Reply:
you can open an issue.